clouddriver-core/src/test/groovy/com/netflix/spinnaker/clouddriver/data/ConfigFileServiceTest.java

```diff
@@ -0,0 +1,154 @@
+  private static final String TEST_FILE_CONTENTS = "test file contents";
+
+  private ResourceRepository resourceRepository = mock(ResourceRepository.class);
+  private EnvironmentRepository environmentRepository = mock(EnvironmentRepository.class);
+
+  private ConfigFileService configFileService;
+
+  @BeforeEach
+  void setUp() {
+    configFileService = new ConfigFileService(resourceRepository, environmentRepository);
+  }
+
+  @AfterEach
+  void tearDown() throws IOException {
+    Files.deleteIfExists(Paths.get(TEST_FILE_NAME));
+  }
+
+  @Test
+  void getLocalPathWhenFileExists() throws IOException {
+    createExpectedFile();
+
+    String fileName = configFileService.getLocalPath(TEST_FILE_NAME, "test", ".file");
+    assertThat(fileName).isEqualTo(TEST_FILE_NAME);
+  }
+
+  @Test
+  void getLocalPathWhenFileDoesNotExist() {
+    RuntimeException exception =
+        assertThrows(
+            RuntimeException.class,
+            () -> configFileService.getLocalPath(TEST_FILE_NAME, "test", ".file"));
+    assertThat(exception.getMessage()).contains(TEST_FILE_NAME);
+  }
+
+  @Test
+  void getLocalPathWhenFileInConfigServer() {
+    expectFileInConfigServer();
+
+    String fileName = configFileService.getLocalPath(CLOUD_TEST_FILE_NAME, "test", ".file");
+    assertThat(baseName(fileName)).startsWith("test").endsWith(".file");
+  }
+
+  @Test
+  void getLocalPathWhenFileNotInConfigServer() {
+    expectFileNotInConfigServer();
+
+    RuntimeException exception =
+        assertThrows(
+            RuntimeException.class,
+            () -> configFileService.getLocalPath(CLOUD_TEST_FILE_NAME, "test", ".file"));
+    assertThat(exception.getMessage()).contains(CLOUD_TEST_FILE_NAME);
+  }
+
+  @Test
+  void getLocalPathWhenContentProvided() {
+    String fileName =
+        configFileService.getLocalPathForContents(TEST_FILE_CONTENTS, "test", ".file");
+    assertThat(baseName(fileName)).startsWith("test").endsWith(".file");
+  }
+
+  @Test
+  void getContentsWhenFileExists() throws IOException {
+    createExpectedFile();
+
+    String contents = configFileService.getContents(TEST_FILE_NAME);
+    assertThat(contents).isEqualTo(TEST_FILE_CONTENTS);
+  }
+
+  @Test
+  void getContentsWhenFileInConfigServer() {
+    expectFileInConfigServer();
+
+    String contents = configFileService.getContents(CLOUD_TEST_FILE_NAME);
+    assertThat(contents).isEqualTo(TEST_FILE_CONTENTS);
+  }
+
+  @Test
+  void getContentsWhenFileNotInConfigServer() {
+    expectFileNotInConfigServer();
+
+    RuntimeException exception =
+        assertThrows(
+            RuntimeException.class, () -> configFileService.getContents(CLOUD_TEST_FILE_NAME));
+    assertThat(exception.getMessage()).contains(CLOUD_TEST_FILE_NAME);
+  }
+
+  private String baseName(String fileName) {
+    return new File(fileName).getName();
+  }
+
+  private void createExpectedFile() throws IOException {
+    File file = new File(TEST_FILE_NAME);
```

Comment:
Would it be better to write this file into the configured temp directory instead of the current working directory?

Reply:
Resolved in 2nd commit.